fem/intrules.hpp

```diff
@@ -255,6 +266,26 @@ class IntegrationRule : public Array<IntegrationPoint>
        a call like this: `IntPoint(i).weight`. */
    const Array<double> &GetWeights() const;
 
+   /// Returns true if the two IntegrationRules are equal.
+   /** Different memory addresses will always result in false. */
+   bool operator==(const IntegrationRule &rhs) const
+   {
+      if (this==&rhs)
+      {
+         return static_cast<Array<IntegrationPoint>>(*this) ==
+                static_cast<Array<IntegrationPoint>>(rhs);
+      }
+      else
+      {
+         return false;
+      }
+   }
```

Comment:
I think this might be equivalent to 
```cpp
bool operator==(const IntegrationRule & other) const {
    return (this != &other) && Array<IntegrationPoint>::operator==(other);
}
```

Reply:
There is no `Array<IntegrationPoint>::operator==`, that's why I'm casting, this is defined outside the class as a free function.

Reply:
Ah, okay

Reply:
I removed this method, as @pazner pointed out it wasn't making much sense, it's using `Array<T>`'s `operator==` now.